include/grpc/grpc_security.h

```diff
@@ -768,6 +767,12 @@ typedef struct grpc_tls_credentials_options grpc_tls_credentials_options;
  */
 typedef struct grpc_tls_certificate_provider grpc_tls_certificate_provider;
 
+/* Status generated while performing TLS handshakes. */
```

Comment:
Do we really need this struct? Why not just append the code and error details to function signature?
```
GRPCAPI void grpc_tls_certificate_provider_data_watcher_set_key_cert_pairs(grpc_tls_certificate_provider* provider, grpc_tls_identity_pairs* pem_key_cert_pairs, grpc_status_code* code, const char** error_details)
```
This way we can make it the caller's responsibility to free error_details, and we don't have to create a release API just to free error_details. 
The current structure would have been useful if we had plans on expanding grpc_tls_status, but I don't think that's the case.

@itsemmanuelfrancis @ZhenLian Wdyt?
@johann1000 FYI Since I made a similar comment in certificate-key match PR.

Reply:
I think it's a nice structure for converting between `absl::Status` and `grpc::Status` and that it's very likely it will be reused given how useful `absl::Status` is in gRPC.

Reply:
I do see the upside to letting the caller allocate and free memory too. Thoughts? @ZhenLian

Reply:
Thanks for updating the API.